backend/src/controllers/goal.controller.ts

```diff
@@ -82,6 +80,15 @@ export const getGoal = async (
           res.sendStatus(httpStatus.CONFLICT).end();
         }
       }
+      const userEncounters = JSON.parse(JSON.stringify(userCurrent?.encounters));
+      let progress_count = 0;
+      userEncounters.forEach((encounter) => {
+        if (encounter.date > goalDTO.date_start && encounter.date < goalDTO.date_end) {
+          progress_count++;
+        }
+        return progress_count;
+      });
```

Comment:
This doesn't work because encounter is just an id not an object 

Please type line 83 like this so you get type hints and don't perform unneeded computations
```js
const userEncounters = userCurrent?.encounters ?? [];
```